strongbox-security/strongbox-authentication-providers/strongbox-ldap-authentication-provider/src/main/java/org/carlspring/strongbox/authentication/api/ldap/LdapAuthenticationConfigurationManager.java

```diff
@@ -67,7 +69,7 @@ public void testConfiguration(String username,
             LdapUserDetailsService luds = (LdapUserDetailsService) c.getBean("ldapUserDetailsService");
 
             UserDetails user = luds.loadUserByUsername(username);
-            if (!password.equals(user.getPassword()))
+            if (!PasswordEncoderFactories.createDelegatingPasswordEncoder().matches(password, user.getPassword()))
```

Comment:
@anki2189 I mean that we should use `PasswordEncoderFactories.createDelegatingPasswordEncoder()` instead of `BCryptPasswordEncoder` in `StrongboxSecurityConfig.passwordEncoder()`. This will aslo affect on strongbox users, so we should store passwords with prefix in our DB as well.

Reply:
We should create new task for those changes. What are your thoughts?

Reply:
I will create a task related with #1614 , but changes should be done in this PR because it directly related with what you doing here.

Reply:
Changes done. Pls review :)

Reply:
Sergey, what are your thoughts on storing the algorithm in the database as well? It looks like the default implementation will require that in order to work (check the `*strongbox-security-users.yaml` file changes)? Should we create a custom [`PasswordEncoderFactory`](https://dzone.com/articles/password-encoder-migration-with-spring-security-5) (check `DefaultPasswordEncoderFactories`) to avoid this or leave it as is?

Reply:
yeah, I've expected this and it should be good feature for us. For example if we will decide to change default encryption alg for our passwords then we can see the users which we need to request to change the passwords.